nifi-commons/nifi-utils/src/main/java/org/apache/nifi/util/file/classloader/ClassLoaderUtils.java

```diff
@@ -21,45 +21,59 @@
 import java.net.MalformedURLException;
 import java.net.URL;
 import java.net.URLClassLoader;
+import java.util.Arrays;
 import java.util.LinkedList;
 import java.util.List;
+import java.util.stream.Collectors;
 
 public class ClassLoaderUtils {
 
     public static ClassLoader getCustomClassLoader(String modulePath, ClassLoader parentClassLoader, FilenameFilter filenameFilter) throws MalformedURLException {
-        String[] modules = modulePath != null? modulePath.split(",") : null;
-        URL[] classpaths = getURLsForClasspath(modules,filenameFilter);
-        return createModuleClassLoader(classpaths,parentClassLoader);
+        // Split and trim the module path(s)
+        List<String> modules = (modulePath == null)
+                ? null
+                : Arrays.stream(modulePath.split(",")).map((path) -> path == null ? null : path.trim()).filter((item) -> item != null).collect(Collectors.toList());
```

Comment:
Good point. I changed to use yours (replacing some things with method references) and added a unit test around the errors you found.